web/packages/teleport/src/Discover/SelectResource/SelectResource.tsx

```diff
@@ -82,14 +82,12 @@ export function SelectResource({ onSelect }: SelectResourceProps) {
 
   function onSearch(s: string, customList?: ResourceSpec[]) {
     const list = customList || defaultResources;
-    const split = s.split(' ').map(s => s.toLowerCase());
-    const foundResources = list.filter(r => {
-      const match = split.every(s => r.keywords.includes(s));
-      if (match) {
-        return r;
-      }
-    });
-    setResources(foundResources);
+    const search = s.split(' ').map(s => s.toLowerCase());
+    const found = list.filter(r =>
+      search.every(s => r.keywords?.some?.(k => k.toLowerCase().includes(s)))
```

Comment:
is checking for keywords necessary? we do say with types that it's required 🤔

Reply:
Shouldn't be... is there a downside to keeping this optional chaining in?